### ISSUE TITLE
returnChanges populate changes on error

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#4208]
With 4 documents with the primary 1,2,3 and 4:

```
r.db(...).table(...).replace(function(doc) {
  return {id: 1, foo: doc('foo').add(1)}
}, {returnChanges: true})


// It returns:
{
  "changes": [
    {
      "new_val": {
        "bar": 0,
        "copyId": 2,
        "foo": 2,
        "id": 2
      },
      "old_val": {
        "bar": 0,
        "copyId": 2,
        "foo": 2,
        "id": 2
      }
    },
    {
      "new_val": {
        "bar": 0,
        "copyId": 3,
        "foo": 2,
        "id": 3
      },
      "old_val": {
        "bar": 0,
        "copyId": 3,
        "foo": 2,
        "id": 3
      }
    },
    {
      "new_val": {
        "bar": 0,
        "copyId": 4,
        "foo": 2,
        "id": 4
      },
      "old_val": {
        "bar": 0,
        "copyId": 4,
        "foo": 2,
        "id": 4
      }
    },
    {
      "new_val": {
        "foo": 204,
        "id": 1
      },
      "old_val": {
        "foo": 203,
        "id": 1
      }
    }
  ],
  "deleted": 0,
  "errors": 3,
  "inserted": 0,
  "replaced": 1,
  "skipped": 0,
  "unchanged": 0
}

```

**Comment by neumino** [comment]
Sorry, I read the wrong line again. That's the output of reqlite -_-

**Comment by neumino** [comment]
Actually no, this is really the result returned by RethinkDB.
Since #4068 is merged, it may not be woth to fix it, so feel free to close this issue.

**Comment by mlucy** [comment]
I actually ran into this while fixing https://github.com/rethinkdb/rethinkdb/issues/4068 and fixed it in the same code review (i.e. `return_changes: true` no longer returns changes on an error, but `return_changes: 'always' does).  It should be fixed when 2.0.2 is released.

Thanks for the catch!
